src/platform/packages/shared/kbn-esql-validation-autocomplete/src/shared/helpers.ts

```diff
@@ -719,94 +693,6 @@ export const isParametrized = (node: ESQLProperNode): boolean => Walker.params(n
  */
 export const noCaseCompare = (a: string, b: string) => a.toLowerCase() === b.toLowerCase();
 
-/**
- * This function returns a list of closing brackets that can be appended to 
- * a partial query to make it valid.
-
-* locally fix the queryString to generate a valid AST
- * A known limitation of this is that is not aware of commas "," or pipes "|"
- * so it is not yet helpful on a multiple commands errors (a workaround it to pass each command here...)
- * @param text
- * @returns
- */
-export function getBracketsToClose(text: string) {
-  const stack = [];
-  const pairs: Record<string, string> = { '"""': '"""', '/*': '*/', '(': ')', '[': ']', '"': '"' };
-  const pairsReversed: Record<string, string> = {
-    '"""': '"""',
-    '*/': '/*',
-    ')': '(',
-    ']': '[',
-    '"': '"',
-  };
-
-  for (let i = 0; i < text.length; i++) {
-    for (const openBracket in pairs) {
-      if (!Object.hasOwn(pairs, openBracket)) {
-        continue;
-      }
-
-      const substr = text.slice(i, i + openBracket.length);
-      if (pairsReversed[substr] && pairsReversed[substr] === stack[stack.length - 1]) {
-        stack.pop();
-        break;
-      } else if (substr === openBracket) {
-        stack.push(substr);
-        break;
-      }
-    }
-  }
-  return stack.reverse().map((bracket) => pairs[bracket]);
-}
-
-/**
- * This function counts the number of unclosed parentheses
- * @param text
- */
-export function countUnclosedParens(text: string) {
-  let unclosedCount = 0;
-  for (let i = 0; i < text.length; i++) {
-    if (text[i] === ')' && unclosedCount > 0) {
-      unclosedCount--;
-    } else if (text[i] === '(') {
-      unclosedCount++;
-    }
-  }
-  return unclosedCount;
-}
-
-/**
- * This function attempts to correct the syntax of a partial query to make it valid.
- *
- * This is important because a syntactically-invalid query will not generate a good AST.
- *
- * @param _query
- * @param context
- * @returns
- */
-export function correctQuerySyntax(_query: string, context: EditorContext) {
-  let query = _query;
-  // check if all brackets are closed, otherwise close them
-  const bracketsToAppend = getBracketsToClose(query);
-  const unclosedRoundBracketCount = bracketsToAppend.filter((bracket) => bracket === ')').length;
-  // if it's a comma by the user or a forced trigger by a function argument suggestion
-  // add a marker to make the expression still valid
-  const charThatNeedMarkers = [',', ':'];
-  if (
-    (context.triggerCharacter && charThatNeedMarkers.includes(context.triggerCharacter)) ||
-    // monaco.editor.CompletionTriggerKind['Invoke'] === 0
-    (context.triggerKind === 0 && unclosedRoundBracketCount === 0) ||
```

Comment:
This was an opaque check which strangely added the marker when all the brackets were closed and the trigger kind was "Invoke" (which is most of the time).